lib/shared/s3-packages.js

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright 2016 resin.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const _ = require('lodash');
+const semver = require('semver');
+const Bluebird = require('bluebird');
+const request = Bluebird.promisifyAll(require('request'));
+const xml = Bluebird.promisifyAll(require('xml2js'));
+
+/**
+ * @summary Etcher production releases S3 bucket URL
+ * @constant
+ * @public
+ * @type {String}
+ */
+exports.BUCKET_URL_PRODUCTION = 'https://resin-production-downloads.s3.amazonaws.com';
+
+/**
+ * @summary Etcher releases S3 package name
+ * @constant
+ * @private
+ * @type {String}
+ */
+const S3_PACKAGE_NAME = 'etcher';
+
+/**
+ * @summary Number of packages per Etcher version
+ * @constant
+ * @private
+ * @type {Number}
+ */
+const NUMBER_OF_PACKAGES = 8;
```

Comment:
This is a fantastic idea. Probably not worth doing it in this PR, but I'll add it to my TODOs and will send a PR soon.